code/game/turfs/space/transit.dm

```diff
@@ -11,7 +11,7 @@
 	var/max = world.maxx-TRANSITIONEDGE
 	var/min = 1+TRANSITIONEDGE
 
-	var/_z = rand(ZLEVEL_SPACEMIN,ZLEVEL_SPACEMAX)	//select a random space zlevel
+	var/_z = pick(accessible_z_levels)	//select a random space zlevel

```

Comment:
If this picks an occupied z, it's quite possible you'll literally land in the middle of the station or worse, the asteroid. There's a reason this dumped you off into space.